source/agora/common/BlockStorage.d

```diff
@@ -758,3 +758,253 @@ public size_t readSizeType (File file) @trusted
     file.rawRead(buffer);
     return *cast(size_t*)&(buffer[0]);
 }
+
+
+/*******************************************************************************
+
+    Define the Memory Storage for Blocks
+
+    Implemented using only memory without file IO.
+
+*******************************************************************************/
+
+public class MemoryStorage : IBlockStorage
```

Comment:
`MemoryStorage` is too generic of a name. I would rename it to something like `MemBlockStorage` instead.